src/coreclr/tools/Common/Compiler/GenericCycleDetection/ModuleCycleInfo.cs

```diff
@@ -142,7 +142,20 @@ private bool IsDeepPossiblyCyclicInstantiation(TypeDesc type, ref int breadthCou
                 {
                     case TypeFlags.Array:
                     case TypeFlags.SzArray:
-                        return IsDeepPossiblyCyclicInstantiation(((ParameterizedType)type).ParameterType, ref breadthCounter, seenTypes);
+                        TypeDesc parameterType = type;
+                        int arrayNesting = 0;
+                        do
+                        {
+                            parameterType = ((ParameterizedType)parameterType).ParameterType;
+                            arrayNesting++;
+                        } while (parameterType.IsArray);
+
+                        if (arrayNesting > _depthCutoff)
```

Comment:
Unlike the generic case, this does not check for cycles. Is this going to fire for non-recursive cases too?

Reply:
> Unlike the generic case, this does not check for cycles. Is this going to fire for non-recursive cases too?

We don't call IsDeepPossiblyCyclicInstantiation unless FormsCycle was true:

https://github.com/dotnet/runtime/blob/551379a185a011789e29a9d326804eb15995f2c7/src/coreclr/tools/Common/Compiler/GenericCycleDetection/ModuleCycleInfo.cs#L246-L250

For normal apps we don't run this while loop at all. I don't see the generic case below having additional validation either. We're dealing with something that has a flagged formal, we're too deep, we cut it off.